File: 02 - ScoreBoard/backend/app/models/team.py
from typing import List, Literal, Optional

from .common import CamelModel

CompetitionScope = Literal["domestic", "european", "international"]


class TeamCompetitionLink(CamelModel):
    league_id: str
    scope: CompetitionScope


class Team(CamelModel):
    id: str
    name: str
    short_name: str
    crest_url: Optional[str] = None
    country: str
    competitions: List[TeamCompetitionLink]
    is_popular: Optional[bool] = None

File: 02 - ScoreBoard/frontend/src/types/team.ts
export type CompetitionScope = "domestic" | "european" | "international";

export interface TeamCompetitionLink {
  leagueId: string;
  scope: CompetitionScope;
}

export interface Team {
  id: string;
  name: string;
  shortName: string;
  crestUrl: string | null;
  country: string;
  /** Leagues/competitions this team currently participates in. */
  competitions: TeamCompetitionLink[];
  /** Whether this team should be surfaced in "popular matches" on Home. */
  isPopular?: boolean;
}
